alns/stopping_criteria/MaxRuntime.py

```diff
@@ -0,0 +1,41 @@
+import time
+
+from alns.State import State
+from alns.stopping_criteria.StoppingCriterion import StoppingCriterion
+
+
+class MaxRuntime(StoppingCriterion):
+    def __init__(self, max_runtime: int) -> None:
+        """
+        Criterion that stops after a maximum number of iterations.
+        """
+        if max_runtime < 0:
+            raise ValueError("Max iterations must be non-negative.")
+
+        self._max_runtime = max_runtime
+        self._elapsed_runtime = None
+        self._start_runtime = None
+
+    @property
+    def max_runtime(self) -> float:
+        return self._max_runtime
+
+    @property
+    def elapsed_runtime(self) -> float:
+        return self._elapsed_runtime
+
+    @property
+    def start_runtime(self) -> float:
+        """
+        Reference point to calculate the elapsed time.
+        """
+        if self._start_runtime is None:
+            self._start_runtime = time.perf_counter()
+
+        return self._start_runtime
```

Comment:
Refactor: similar to `current_iteration` in `MaxIterations`, I am unsure what exposing these fields on a particular StoppingCriterion adds to the library. In order:

- `elapsed_time`: that's equivalent to `sum(res.statistics.runtimes)`. So this is already available for every StoppingCriterion through the Result object.
- `start_runtime`. That is more or less `time.perf_counter() - sum(res.statistics.runtime)`. So this is already available for every StoppingCriterion through the Result object.

A better idea might be to make these properties available on the Result object, so they can always be accessed.

---

In `MaxRuntimes`, you can then set `_start_runtime` the first time `__call__` is called. That should also make the comment there unnecessary! :-)

Reply:
`start_runtime` should probably be passed from `ALNS.iterate` as an argument to `Result`. Maybe by setting `start = time.perf_counter()` near the top of `ALNS.iterate`.

Reply:
What is the added benefit of having `start_runtime` available to `Result` (or did you mean `Statistics`?)?

Reply:
What might be a nice addition is to implement `statistics.total_runtime` as property. What do you think?

Reply:
>Moreover, I believe that statistics._runtimes[0] is equivalent to the start time.

It is. We should probably expose that as `Statistics.start_time`.

>What might be a nice addition is to implement statistics.total_runtime as property. What do you think?

Agreed! :-)

Reply:
See https://github.com/N-Wouda/ALNS/pull/64/commits/0c0de8e6cb3c55e3ee8b95f1bfcd4db4eb5dd2a2